os/android/gralloc1bufferhandler.cpp

```diff
@@ -129,15 +133,43 @@ void Gralloc1BufferHandler::CopyHandle(HWCNativeHandle source,
   CopyBufferHandle(source, target);
 }
 
-void *Gralloc1BufferHandler::Map(HWCNativeHandle /*handle*/, uint32_t /*x*/,
-                                 uint32_t /*y*/, uint32_t /*width*/,
-                                 uint32_t /*height*/, uint32_t * /*stride*/,
-                                 void ** /*map_data*/, size_t /*plane*/) {
+void *Gralloc1BufferHandler::Map(HWCNativeHandle handle, uint32_t x,
+                                 uint32_t y, uint32_t width,
+                                 uint32_t height, uint32_t * /*stride*/,
+                                 void ** map_data, size_t /*plane*/) {
+  auto gr_handle = (struct cros_gralloc_handle *)handle->imported_handle_;
+  if (!gr_handle) {
+    ETRACE("could not find gralloc drm handle");
+    return NULL;
+  }
+
+  int acquireFence = -1;
+  gralloc1_rect_t rect{};
+  rect.left = x;
+  rect.top = y;
+  rect.width = width;
+  rect.height = height;
+
+  gralloc1_device_t *gralloc1_dvc =
+      reinterpret_cast<gralloc1_device_t *>(device_);
+  uint32_t status = lock_(gralloc1_dvc, handle->imported_handle_, GRALLOC_USAGE_SW_READ_RARELY,
+          GRALLOC_USAGE_SW_READ_RARELY, &rect, map_data, acquireFence);
+  ALOGI("%s - lock returned: %u", __FUNCTION__, status);
   return NULL;
```

Comment:
retrun NULL here is confusing, probably we should share the API ?

Reply:
yeah, we can just return the status instead.

Reply:
I think the map api is to return the vritual interface CPU can access? That is at leat the story in Linux side. Linux is directly this API to gbm_bo_map